platform/src/main/java/org/hillview/main/BatchLogAnalysis.java

```
@@ -0,0 +1,188 @@
+        int numOfBucketsD2 = heatmap.getNumOfBucketsD2();
+        heatmapData.matrix = new long[numOfBucketsD1][numOfBucketsD2];
+        for (int i = 0; i<numOfBucketsD1; i++){
+            for (int j = 0; j<numOfBucketsD2; j++){
+                heatmapData.matrix[i][j] = heatmap.getCount(i, j);
+            }
+        }
+        heatmapData.errorCodeLabels = leftBoundaries;
+        for (int x = 0; x < numOfTimestampBuckets; x++){
+            double time = dataRange.min + x * (dataRange.max - dataRange.min) / numOfTimestampBuckets;
+            Instant instantTime = Converters.toDate(time);
+            String stringDate = Converters.toString(instantTime);
+            heatmapData.timeLabels.add(stringDate);
+        }
+        return heatmapData;
+    }
+
+    private static void saveHeatmapToFile(HeatmapData heatmapData, String filepath) {
+        try (PrintWriter writer = new PrintWriter(new File(filepath))) {
+            StringBuilder sb = new StringBuilder();
+            sb.append("Timestamp");
+            sb.append(',');
+            sb.append("errorCode");
+            sb.append(',');
+            sb.append("Count");
+            sb.append('\n');
+            for (int x = 0; x < heatmapData.matrix.length; x++) {
+                for (int y = 0; y < heatmapData.matrix[0].length; y++) {
+                    if (heatmapData.matrix[x][y] >= 0) {        // keep zeros or not
+                        sb.append(heatmapData.timeLabels.get(x));
+                        sb.append(',');
+                        sb.append(heatmapData.errorCodeLabels.get(y).replaceAll(",",""));
+                        sb.append(',');
+                        sb.append(heatmapData.matrix[x][y]);
+                        sb.append('\n');
+                    }
+                }
+            }
+            writer.write(sb.toString());
+        } catch (FileNotFoundException e) {
+            System.out.println(e.getMessage());
```

It;s usually not a good idea to completely swallow an exception.
Why don't you just stop processing in this case?